### DISCUSSION TITLE
Is there an example of drawing clipped text?

### DISCUSSION THREAD

**Discussion by mshakurov** [SixLabors/ImageSharp.Drawing#328] [Q&A] [answered]
Sorry, is there a simple and clear example of how to draw text in a clipped rectangle?
There is a query [#19](https://github.com/SixLabors/ImageSharp.Drawing/issues/19) with a similar question, but there is no example given. Due to the lack of examples, I've tried a variety of ways, and I can't draw the text clipped in a rectangle.
Task: it is necessary to draw text that does not go beyond the specified rectangle, while not changing the background within this rectangle.

**Comment by tocsoft** [comment, marked as answer]
your looking for the `Clip` processor. You use it by passing any polygon as the first argument and then what ever processing you want applied inside that polygon as the 2nd.

 `img.Mutate(i=>i.Clip(new RectangularPolygon(10, 10, 100, 100) , j =>j.DrawText(...)))`

Here is the sample from out samples repo https://github.com/SixLabors/Samples/blob/main/ImageSharp/ApplyEffectInsideShape/Program.cs for this api.
